app.js

```diff
@@ -1,49 +1,66 @@
-let booksArray = [];
-const localDataStorage = JSON.parse(localStorage.getItem('booksArray'));
-const BookListContainer = document.querySelector('.book-section');
+class BookList {
+  constructor() {
+    this.books = JSON.parse(localStorage.getItem('books')) || [];
+    this.table = document.querySelector('#book-list tbody');
+    this.form = document.querySelector('form');
+    this.titleInput = document.querySelector('#title');
+    this.authorInput = document.querySelector('#author');
 
-if (localDataStorage != null) {
-  localDataStorage.forEach((bookItem) => {
-    BookListContainer.innerHTML += `
-  <div>
-  <div><strong>${bookItem.name}</div>
-  <div>${bookItem.author}</strong></div>
-  <button class='remove-btn' id='${bookItem.id}' 
-    onclick='deleteBook(${bookItem.id})'>Remove</button>
-  </div>
-  <hr>
-  `;
-    const book = {
-      id: bookItem.id,
-      name: bookItem.name,
-      author: bookItem.author,
-    };
-    booksArray.push(book);
-  });
-}
+    // Bind event listeners
+    this.form.addEventListener('submit', this.addBook.bind(this));
+    this.table.addEventListener('click', this.removeBook.bind(this));
+    // Display books
+    this.displayBooks();
+  }
 
-const btnSubmit = document.querySelector('#add-btn');
-const BookName = document.querySelector('#title');
-const BookAuthor = document.querySelector('#author');
+  // Save books to local storage
+  saveBooks() {
+    localStorage.setItem('books', JSON.stringify(this.books));
+  }
 
-// ADD Book Funciton
-function addBook() {
-  const book = {
-    id: booksArray.length,
-    name: BookName.value,
-    author: BookAuthor.value,
-  };
-  booksArray.push(book);
-  localStorage.setItem('booksArray', JSON.stringify(booksArray));
-}
+  // Display all books in the list
+  displayBooks() {
+    this.table.innerHTML = '';
+    this.books.forEach((book) => {
+      const row = document.createElement('tr');
+      row.className = 'row';
+      row.innerHTML = `
+        <td class="content"><strong>${book.title}</strong><em> &ensp;by&ensp; </em>  <strong>${book.author}<strong></td> 
+        <td class ="delete-btn" ><button class="delete"><strong class="delete-button">Remove</strong></button></td>
+      `;
+      this.table.appendChild(row);
+    });
+  }
 
-btnSubmit.addEventListener('click', addBook);
-const remove = document.querySelector('.remove-btn');
+  // Add a new book to the list
+  addBook(e) {
+    e.preventDefault();
+    const title = this.titleInput.value.trim();
+    const author = this.authorInput.value.trim();
```

Comment:
- [ **OPTIONAL** ] I think it would be better if you add an `id` property in the book object so it can help you to know which book you delete  if all books have the same `author_name` and `title_name`. You can use the `Date.now()` method to so.
```JavaScript
const bookID = Date.now()
...
```